### PR TITLE
fix Issue 22825 - #line parsing doesn't follow the spec

### PR DESCRIPTION
**Pull request by ibuclaw**
Because these special token sequences are sensitive to newlines, they should not be allowed after the introducing `#` token.

### CODE REVIEW

File: src/dmd/lexer.d [hunk 1]
```diff
@@ -201,8 +202,22 @@ class Lexer
...
+    /****************************
+     * Turn next token in buffer into a token, including newlines.
+     * Params:
+     *  t = the token to set the resulting Token to
+     */
+    final void scanUntilNewline(Token* t)
+    {
+        this.tokenizeNewlines = true;
+        scan(t);
+        this.tokenizeNewlines = false;
```
Comment (ibuclaw) [inline src/dmd/lexer.d:214]: Bikeshed topic, whether to just make this field a parameter of `scan()` or not.  I'm on the fence either way.

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/lexer.d [hunk 2]
```diff
@@ -201,8 +202,22 @@ class Lexer
...
+    /****************************
+     * Turn next token in buffer into a token, including newlines.
+     * Params:
+     *  t = the token to set the resulting Token to
+     */
+    final void scanUntilNewline(Token* t)
+    {
+        this.tokenizeNewlines++;
+        scan(t);
+        this.tokenizeNewlines--;
```
Comment (ibuclaw) [inline src/dmd/lexer.d:214]: Bikeshed topic, whether to just make this field a `bool` parameter of scan() or not. I'm on the fence either way.

**Review by WalterBright — CHANGES_REQUESTED** [review]
I think this problem will take care of itself if you implement the changes requested in https://github.com/dlang/dmd/pull/13728

**Comment by ibuclaw** [comment]
> I think this problem will take care of itself if you implement the changes requested in #13728

#13728 only take care of `# line` because of shared code path in `case '#'`, but everything else that follows would still be in violation.  As I said, this PR modifies `poundLine`, the other modifies `pragmaPack`.

**Comment by ibuclaw** [comment]
Having read your comments, I see what you mean though, we'll wait and see after (or if) the other PR has been accepted though.

**Comment by ibuclaw** [comment]
In the meantime, added a spec PR here dlang/dlang.org#3234

**Comment by WalterBright** [comment]
@ibuclaw needs rebase!

**Comment by ibuclaw** [comment]
Tests moved to #13728.

**Comment by ibuclaw** [comment]
Tests moved to #13728.

**Pull request closed without merging**